### PR TITLE
[flink] MySqlSyncTableAction now supports flink run

### PR DESCRIPTION
**Pull request by tsreaper**
### Purpose

`MySqlSyncTableAction` now supports being used from `bin/flink run`. Users can use `MySqlSyncTableAction` without writing and compiling their own data stream jobs.

### Tests

* MySql57E2eTest
* MySql80E2eTest

### API and Format 

N/A

### Documentation

Document will be introduced in the next PR.

### CODE REVIEW

File: paimon-flink/paimon-flink-1.14/src/main/java/org/apache/paimon/flink/utils/SingleOutputStreamOperatorUtils.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Utility methods for {@link SingleOutputStreamOperator}.
+ *
+ * <p>TODO: merge all Flink version related utils into one class.
+ */
+public class SingleOutputStreamOperatorUtils {
+
+    public static <T> DataStream<T> getSideOutput(
+            SingleOutputStreamOperator<?> input, OutputTag<T> outputTag) {
+        return input.getSideOutput(outputTag);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-1.14/src/main/java/org/apache/paimon/flink/utils/SingleOutputStreamOperatorUtils.java:34]: What is different between 1.14 and 1.17?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncTableAction.java [hunk 2]
```diff
@@ -118,13 +123,13 @@ public class MySqlSyncTableAction implements Action {
...
     }
 
-    @Override
-    public void run() throws Exception {
-        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-        build(env);
-        env.execute(String.format("MySQL CTAS: %s.%s", database, table));
+    private void setDefaultMySqlConfig() {
+        // https://ververica.github.io/flink-cdc-connectors/master/content/connectors/mysql-cdc.html#connector-options
+        mySqlConfig.putIfAbsent("port", "3306");
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncTableAction.java:132]: Can we convert `mySqlConfig` to Flink `Configuration`?
Then we can get from it directly.

**Review by JingsongLi — COMMENTED** [review]
Left two comments.

File: paimon-flink/paimon-flink-1.14/src/main/java/org/apache/paimon/flink/utils/SingleOutputStreamOperatorUtils.java (hunk 1 above)
Comment (tsreaper) [reply paimon-flink/paimon-flink-1.14/src/main/java/org/apache/paimon/flink/utils/SingleOutputStreamOperatorUtils.java:34]: Their return types are different. In 1.14 and 1.15, return type of `SingleOutputStreamOperator#getSideOutput` is just `DataStream`, while in 1.16 and 1.17, return type is the newly introduced `SideOutputStream`. This `SideOutputStream` does not exist in 1.14 and 1.15.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**